src/pages/RegPage.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from "react";
+import DisplayRequest from "../views/DisplayRequest";
+import FlowData from "../views/FlowData";
+import API, { getFormattedURL, getFormattedProxyURL } from "../API";
+import RegistrationProvider from "../util/RegistrationProvider";
+import { Container, Row, Col } from "react-bootstrap";
+import Registration from "../views/Registration";
+
+const RegPage = (props) => {
+    const reset = () => {
+        setRequest({});
+        setResponse({});
+        setCSRF("");
+        setFlowID("");
+        setSessionID("");
+    }
+
+    const resetPrint = () => {
+        setRequest({});
+        setResponse({});
+    }
+
+    const [request, setRequest] = useState({});
+
+    const [response, setResponse] = useState({});
+
+    const [flowID, setFlowID] = useState("");
+    const [csrf, setCSRF] = useState("");
+    const [accountKey, setAccountKey] = useState("");
+    const [sessionID, setSessionID] = useState("");
```

Comment:
```suggestion
    const [state, setState] = useState(initialState);
    const updateState = (changes) => setState({...state, ...changes});

    const reset = () => updateState(initialState);
    const resetPrint = () => updateState({ request: {}, response: {} });
```

We use a single object for our state. We define a helper function `updateState`, that updates all keys present in the argument object.

Reply:
smart, I like that a lot